DPGAnalysis/MuonSysAging/plugins/DTChamberMasker.cc

```diff
@@ -0,0 +1,189 @@
+// -*- C++ -*-
+//
+// Package:    DPGAnalysis/MuonSysAging
+// Class:      DTChamberMasker
+// 
+/**\class DTChamberMasker DTChamberMasker.cc DPGAnalysis/MuonSysAging/plugins/DTChamberMasker.cc
+
+ Description: Class to mask DT digis on a chamber by chamber basis
+
+*/
+//
+// Original Author:  Carlo Battilana
+//         Created:  Sun, 11 Jan 2015 15:12:51 GMT
+//
+//
+
+
+// system include files
+#include <map>
+#include <memory>
+#include <string>
+#include <iostream>
+#include <algorithm>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Run.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "Geometry/Records/interface/MuonGeometryRecord.h"
+#include "Geometry/DTGeometry/interface/DTGeometry.h"
+
+#include "Geometry/DTGeometry/interface/DTChamber.h"
+#include "Geometry/DTGeometry/interface/DTLayer.h"
+
+#include "DataFormats/DTDigi/interface/DTDigiCollection.h"
+#include "DataFormats/MuonDetId/interface/DTChamberId.h"
+#include "DataFormats/MuonDetId/interface/DTLayerId.h"
+#include "DataFormats/MuonDetId/interface/DTWireId.h"
+
+#include "CondFormats/MuonSystemAging/interface/MuonSystemAging.h"
+#include "CondFormats/DataRecord/interface/MuonSystemAgingRcd.h"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/RandomNumberGenerator.h"
+#include "CLHEP/Random/RandomEngine.h"
+
+
+//
+// class declaration
+//
+
+class DTChamberMasker : public edm::EDProducer 
```

Comment:
This class needs to be a modern producer (probably an `edm::stream::EDProducer`)